app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/chart/BeaconLineChart.java

```diff
@@ -317,72 +329,83 @@ protected void drawBeacons(Canvas canvas) {
-
-            if (lastLinePoint != null && lastAdvertisingPacket != null) {
-                if (advertisingPacket.getTimestamp() < lastAdvertisingPacket.getTimestamp() + 5000) {
-                    canvas.drawLine(
-                            lastLinePoint.x,
-                            lastLinePoint.y,
-                            currentLinePoint.x,
-                            currentLinePoint.y,
-                            linePaint
-                    );
+            for (AdvertisingPacket advertisingPacket : (List<AdvertisingPacket>) beacon.getAdvertisingPackets()) {
+
+                if (advertisingPacket.getTimestamp() < minimumAdvertisingTimestamp) {
+                    continue;
                 }
-            } else {
-                lastLinePoint = new PointF();
+
+                filter.setMinimumTimestamp(advertisingPacket.getTimestamp() - windowLength);
+                filter.setMaximumTimestamp(advertisingPacket.getTimestamp());
+
+                currentLinePoint = getPointFromAdvertisingPacket(beacon, advertisingPacket, currentLinePoint, filter);
+                canvas.drawCircle(currentLinePoint.x, currentLinePoint.y, pixelsPerDip * 1.5f, linePaint);
+
+                if (lastLinePoint != null && lastAdvertisingPacket != null) {
+                    if (advertisingPacket.getTimestamp() < lastAdvertisingPacket.getTimestamp() + 5000) {
+                        canvas.drawLine(
+                                lastLinePoint.x,
+                                lastLinePoint.y,
+                                currentLinePoint.x,
+                                currentLinePoint.y,
+                                linePaint
+                        );
+                    }
+                } else {
+                    lastLinePoint = new PointF();
+                }
+
+                lastLinePoint.x = currentLinePoint.x;
+                lastLinePoint.y = currentLinePoint.y;
+                lastAdvertisingPacket = advertisingPacket;
             }
 
-            lastLinePoint.x = currentLinePoint.x;
-            lastLinePoint.y = currentLinePoint.y;
-            lastAdvertisingPacket = advertisingPacket;
             currentBeaconIndex++;
-        }
-
-        if (lastLinePoint != null) {
-            linePaint.setAlpha(255);
-            canvas.drawCircle(lastLinePoint.x, lastLinePoint.y, pixelsPerDip * 4, linePaint);
+            if (lastLinePoint != null) {
+                linePaint.setAlpha(255);
+                canvas.drawCircle(lastLinePoint.x, lastLinePoint.y, pixelsPerDip * 4, linePaint);
+            }
         }
     }
 
-    protected float getValue(Beacon beacon, AdvertisingPacket advertisingPacket, long windowLength) {
-        List<AdvertisingPacket> recentAdvertisingPackets = null;
-        int[] recentRssis;
-        float meanRssi;
+    protected float getValue(Beacon beacon, AdvertisingPacket advertisingPacket, long windowLength, RssiFilter filter) {
```

Comment:
No need to pass the `windowLength` here, it's already available in the class scope.